### PR TITLE
Adds schedule for package delete on PackageCloud

### PR DESCRIPTION
**Pull request by gurkanindibay**
Recently, we exceeded our PackageCloud subscription plan storage limits so we decided to delete unnecessary nightly packages from our nightly repositories. I manually deleted the packages for once. With this schedule package deletion is being scheduled so unnecessary package deletion (packages older than 10 days) would be scheduled and we would remove the risk to exceed our storage limits.
Additinally I needed to remove the branch name in the unit test for update_docker since branch is deleted and unnecessary now
By the way you can not see any deleted packages, in the latest execution because packages already deleted in the older executions
You can see successful deletion for community on the execution below
https://github.com/citusdata/tools/runs/4949704514?check_suite_focus=true
For enterprise I deleted the applicable packages while testing so you can see package deletions tomorrow at 15:00 pm Istanbul Time

### CODE REVIEW

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (onurctirtir) [inline packaging_automation/delete_packages_on_packagecloud.py:31]: maybe put 10 to a global variable as in PAGE_RECORD_COUNT:
`PACKAGE_DELETION_DAYS_THRESHOLD = 10`

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (onurctirtir) [inline packaging_automation/delete_packages_on_packagecloud.py:44]: maybe comment:
```py
# no more packages older than PACKAGE_DELETION_DAYS_THRESHOLD
```

File: packaging_automation/tests/test_update_docker.py
Comment (onurctirtir) [inline packaging_automation/tests/test_update_docker.py:27]: let's delete that outdated comment then

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (onurctirtir) [inline packaging_automation/delete_packages_on_packagecloud.py:48]:
```suggestion
    print("Deletion Stats:")
    print(f"\tSuccessful deletions: {successful_count}")
    print(f"\tError deletions: {error_count}")
```

File: .github/workflows/delete-packagecloud-packages.yml
Comment (onurctirtir) [inline .github/workflows/delete-packagecloud-packages.yml:33]:
```suggestion
      - name: Delete Old Nightly PackageCloud Packages
```

File: .github/workflows/delete-packagecloud-packages.yml
Comment (onurctirtir) [inline .github/workflows/delete-packagecloud-packages.yml:11]: I think once a week would be sufficient, since we are anyway deleting packages older than 10 days atm.

Also, maybe leave a link to crontab.guru so that readers can easily understand when this job gets executed: 
```suggestion
      # https://crontab.guru/#0_18_*_*_1
      - cron: "0 18 * * 1"
```

**Review by onurctirtir — COMMENTED** [review]
LGTM after addressing the comments

File: .github/workflows/delete-packagecloud-packages.yml
Comment (hanefi) [inline .github/workflows/delete-packagecloud-packages.yml:14]: Why do we call this a unit test?

File: .github/workflows/delete-packagecloud-packages.yml
Comment (hanefi) [inline .github/workflows/delete-packagecloud-packages.yml:11]: I do not think that on.push.schedule is valid.

```suggestion
on:
  schedule:
    - cron: '0 18 * * *'

```

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (hanefi) [inline packaging_automation/delete_packages_on_packagecloud.py:29]:
```suggestion
            package_upload_date = datetime.strptime(package_info['created_at'],
                                                    "%Y-%m-%dT%H:%M:%S.000Z")
```

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (hanefi) [inline packaging_automation/delete_packages_on_packagecloud.py:53]:
```suggestion
    parser.add_argument('--package_repo',
                        choices=[r.value for r in PackageRepository],
                        required=True)

```

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (hanefi) [inline packaging_automation/delete_packages_on_packagecloud.py:57]:
```suggestion
    delete_packages(repo=PackageRepository(args.package_repo),
                    package_cloud_api_token=args.package_cloud_api_token)
```

**Review by hanefi — CHANGES_REQUESTED** [review]

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py:31]: Fixed as you suggested

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py:44]: Fixed as you suggested

File: packaging_automation/tests/test_update_docker.py
Comment (gurkanindibay) [reply packaging_automation/tests/test_update_docker.py:27]: Fixed as you suggested

File: .github/workflows/delete-packagecloud-packages.yml
Comment (gurkanindibay) [reply .github/workflows/delete-packagecloud-packages.yml:33]: Fixed as you suggested

File: .github/workflows/delete-packagecloud-packages.yml
Comment (gurkanindibay) [reply .github/workflows/delete-packagecloud-packages.yml:14]: copy paste error :)
Fixed

File: .github/workflows/delete-packagecloud-packages.yml
Comment (gurkanindibay) [reply .github/workflows/delete-packagecloud-packages.yml:11]: I added it just to execute for tests. Because executing it when opening new PR it is useful to make sure it isn't broken. If you want I can remove by the way

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py:29]: Fixed as you suggested

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py:53]: Fixed as you suggested

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py:57]: Fixed as you suggested

File: packaging_automation/delete_packages_on_packagecloud.py
Comment (gurkanindibay) [reply packaging_automation/delete_packages_on_packagecloud.py]: Fixed as you suggested

File: .github/workflows/delete-packagecloud-packages.yml
Comment (onurctirtir) [reply .github/workflows/delete-packagecloud-packages.yml:12]: makes sense then

**Review by onurctirtir — COMMENTED** [review]
Looks good to me, I'm fine with merging after addressing Hanefi's comments

File: .github/workflows/delete-packagecloud-packages.yml
Comment (hanefi) [inline .github/workflows/delete-packagecloud-packages.yml:12]:
```suggestion
on:
  push:
    branches:
      - "**"
  schedule:
    # https://crontab.guru/#0_18_*_*_*
    - cron: "0 18 * * *"
```

File: .github/workflows/delete-packagecloud-packages.yml
Comment (hanefi) [reply .github/workflows/delete-packagecloud-packages.yml]: It makes sense to keep `on.push.branches` during development then. You can remove that before merges.

However, `on.push.schedule` is not valid and you have an indentation problem here.

See https://github.com/citusdata/tools/pull/214/files#r794111658 for my suggested fix

File: .github/workflows/delete-packagecloud-packages.yml
Comment (hanefi) [inline .github/workflows/delete-packagecloud-packages.yml:9]: Does it really make sense to run this job on every branch push?

I suggest we remove `on.push` lines, and use only `on.schedule`.

**Review by hanefi — COMMENTED** [review]
Looks good after we fix the yaml definition for the schedule.

Please use a validator and check that new definitions are correct when creating a new yaml file, or updating an existing one.

File: .github/workflows/delete-packagecloud-packages.yml
Comment (gurkanindibay) [reply .github/workflows/delete-packagecloud-packages.yml]: Fixed as you suggested

File: .github/workflows/delete-packagecloud-packages.yml
Comment (gurkanindibay) [reply .github/workflows/delete-packagecloud-packages.yml]: Fixed as you suggested

**Review by hanefi — APPROVED** [review]

**Pull request merged**
